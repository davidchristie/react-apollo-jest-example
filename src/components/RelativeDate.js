import PropTypes from 'prop-types'
import React, { Component } from 'react'
import relativeDate from 'relative-date'

export default class RelativeDate extends Component {
  static propTypes = {
    value: PropTypes.string.isRequired
  }

  render () {
    return (
      <span>
        {relativeDate(new Date(this.props.value))}
      </span>
    )
  }
}
