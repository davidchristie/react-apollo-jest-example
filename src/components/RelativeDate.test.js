import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import RelativeDate from './RelativeDate'

describe('RelativeDate component', () => {
  it('should render correctly', () => {
    expect(toJson(mount(
      <RelativeDate value={new Date().toString()} />
    ))).toMatchSnapshot()
  })
})
