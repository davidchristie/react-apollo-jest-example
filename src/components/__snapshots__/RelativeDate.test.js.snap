// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelativeDate component should render correctly 1`] = `
<RelativeDate
  value="Wed Sep 20 2017 17:33:57 GMT+1200 (NZST)"
>
  <span>
    just now
  </span>
</RelativeDate>
`;
